### ISSUE TITLE
factoree & promises

### ISSUE THREAD

**Issue by kyusu** [Schniz/factoree#2]
Hi @Schniz,

first of all: thanks for `factoree`. It is quite a helpful library and I would not want to do without it when writing unit tests.

There is one thing I stumble upon from time to time: let's say I want to mock the function `getPerson` with the following signature:

```typescript
type Person = {name: string, age: number}
type GetPerson = () => Promise<Person>;
```
and let's pretend the type of property `age` is quite a bit more complicated and does not matter for my test. I would then write something like this:

```typescript
const personFactory = factory<Person>({name: undefined})
const mockedGetPerson = () => Promise.resolve<Person>(personFactory({name: "Gal Schlezinger"}))
```

Which seems fine and compiles. But when I run it, `Promise.resolve` will checks whether the value passed into it, is itself a promise or not. 
A nice `UnhandledPromiseRejectionWarning: Error: Can't access undefined key ''then'' for object { name: 'Gal Schlezinger'}` is thrown. 

This makes sense, as the `proxy` does not know about the key `then`. 

To circumvent this problem, one could try to introduce a `then`:
```typescript
const personFactory = factory<Person>({name: undefined, then: undefined})
```
but the TypeScript compiler rightfully complains about this:
```
 TS2345: Argument of type '{ name: undefined; then: undefined; }' is not assignable to parameter of type 'Partial<Person>'.
  Object literal may only specify known properties, and 'then' does not exist in type 'Partial<Person>'
  ```

The only way around this dilemma I could come up with, was to create a `factory` for the intersection of `T & PromiseLike<T>` instead of just for `T` which lets me define a `then`:

```typescript
const personFactory = factory<Person & PromiseLike<Person>>({name: undefined, then: undefined});

const mockedGetPerson = () => Promise.resolve<Person>(personFactory({name: "Gal Schlezinger"});

```

Do you know whether there is a better solution for this?

Thanks in advance!

**Comment by Schniz** [comment]
Thanks for the kind words!
Seems like we need to add more exceptions to the keys we use to break stuff. This is unfortunate because you'd think that they will use something like:

```
function isThenable(obj) {
  return 'then' in obj && typeof obj.then === 'function'
}
```

We'll fix it in #3 😃

**Comment by Schniz** [comment]
Published 0.1.1 with `Promise.resolve` support 😃

**Comment by kyusu** [comment]
Hi @Schniz,

that was fast! Thank you very much!
